Gemfile

```diff
@@ -10,8 +10,8 @@ end
 
 # In addition to :build, for contributing
 group :development do
-	gem 'cocoapods', '~> 1.6.0'
-	gem 'rubocop', '~> 0.65'
+	gem 'cocoapods', '~> 1.7.0'
```

Comment:
I'd actually suggest that we change that version constraint to `~> 1.8` or even `~> 1.9`

There's no reason to constraint it to a specific `major.minor.x` (and only allow the patch component to be bumped) as opposed to letting the minor version be bumped to. SemVer is all about the fact that there's no breaking change as long as you keep the same major version, and `~> x.y` is recommended over `~> x.y.z`. Not sure why we used 3 digits constraints in the past tbh.

Latest version of CocoaPods is 1.9.x, and has the huge advantage of being able to use the CDN instead of having to clone the whole master Spec repo (which takes much more disk space and is long to clone). The use of the spec CDN improves both disk space and performance of `pod install` and avoids the very long initial "Setting up CocoaPods master repo" step on initial runs. So we might as well use that latest version anyway.

Reply:
I agree with the sem versioning, I blindly followed what was already used there.

Yeah, let's try to go update the Cocoapods version - the issue is that installing ruby `>= 2.4.0` is not so easy to do in Ubuntu as `2.3.x` is. I'll try to create a new docker image, but it's gonna take me some more time.

Reply:
Maybe you can try with one of the official Swift Linux images on [Docker Hub](https://hub.docker.com/_/swift), e.g. `swift:4.2`

It would also be worth updating StencilSwiftKit to Swift 5 first (obviously we should ultimately do that in a separate PR). I think that would incidentally solve a lot of the issues you're having here, because once we'll be updated to use Swift 5, we can point to a `swift:5.2` image in Docker, and it being a more recent image, we'll also have a very good chance that the ruby version would be more recent as well and compatible with what `cocoapods-downloader` expects

Reply:
I actually just did a very quick "Edit > Convert to Current Swift Syntax" in the Xcode project to convert to Swift 5 and apparently, there's not a single change in the source code needed to make it compile with Swift 5. So just changing `s.swift_version` in the `podspec` and re-doing a `bundle exec pod install` so that the corresponding target in Xcode has an updated `SWIFT_VERSION` should probably be enough.

Since we're planning to upgrade to CocoaPods 1.9, we might also as well:
 - Update the `s.cocoapods_version = '>= 1.4.0'` in the `podspec` to a more recent requirement (at least 1.8 if not 1.9)
 - Use the plural form of the parameter `s.swift_versions = ["4.2", "5.0"]` instead of the singular form to indicate that we support both 4.2 and 5.0?

Reply:
Unfortunately, the official Swift Linux image does not have `bundle` installed. What I can do is create a new docker image that will use Swift's image as a base and then installs ruby with the version we need + `bundle`. I tried to find already-existing suitable image, but could not find any.

Reply:
@AliSoftware I'd recommend setting `s.cocoapods_version` to the lowest one allowed (only needed for swift_versions property).

Reply:
Yeah, I've always had trouble navigating Docker Hub to find the image I want, especially knowing what's inside each (swift version AND ruby version)

I think before trying that, it'd be better to convert the project to Swift 5 first (see @djbe 's comment below, there is an open PR already pending that might need refreshing) and retry. Once we use Swift 5 and point to a Docker Image with Swift 5 (like `norionomura/jazzy:swift-5.2.0`) it might actually also already contain the more recent Ruby version we want…

Reply:
I have tried using `norionmure/jazzy:swift-5.2.0` but that still contains ruby 2.3.x. They are even constraining the `cocoapods-downloader` [here](https://github.com/norio-nomura/docker-jazzy/blob/master/swift-5.2.5/Dockerfile#L19). The issue is that `apt-get install -y ruby-dev` does not let you install anything newer than `2.3.x`. 

I'll take a peek to that PR for the Swift version bump.

Reply:
Ah. 😐 

We have a `.ruby-version` file at the root of our repository, that is used when you use `chruby` to switch/install alternative Ruby versions. I'd suggest you bump the version of ruby in that file and try to see if we can install then use `chruby` in Norio's Docker image to install the ruby version we want…